warehouse/routes.py

```diff
@@ -640,6 +640,20 @@ def includeme(config):
         domain=warehouse,
     )
 
+    config.add_route(
+        "legacy.api.json.user",
+        "/user/{username}/json",
+        factory="warehouse.legacy.api.json.user_factory",
+        domain=warehouse,
+    )
+
+    config.add_route(
+        "legacy.api.json.user_slash",
+        "/user/{username}/json/",
+        factory="warehouse.legacy.api.json.user_factory",
+        domain=warehouse,
+    )
```

Comment:
Having a dedicates route for the slash variant seems a little funky (versus redirecting), but I see it's what we do for the other routes under `legacy.api`. Curious if @miketheman or @di has thoughts about this 🙂

Reply:
Agreed. For now I've just done what the other routes are doing but something cleaner would be nice.